lms/djangoapps/learner_dashboard/tests/test_programs.py

```diff
@@ -320,3 +327,11 @@ def test_page_routing(self):
 
         response = self.client.get(self.details_page + 'program_name/invalid/')
         self.assertEquals(response.status_code, 404)
+
+    def test_404_if_no_data(self):
+        """Verify that the page 404s if no program data is found."""
+        self.create_programs_config()
+        self._mock_programs_api(status=404)
+
+        response = self.client.get(self.details_page)
+        self.assertEquals(response.status_code, 404)
```

Comment:
Can you add one more test case to have the Program return actual empty data with a 200 status code?